crates/wasi-provider/src/handle.rs

```diff
@@ -46,25 +52,125 @@ impl<R: AsyncReadExt + AsyncSeekExt + Unpin> RuntimeHandle<R> {
     /// Signal the running instance to stop and wait for it to complete. As of
     /// right now, there is not a way to do this in wasmtime, so this does
     /// nothing
-    pub async fn stop(&mut self) -> anyhow::Result<()> {
+    pub(crate) async fn stop(&mut self) -> anyhow::Result<()> {
         // TODO: Send an actual stop signal once there is support in wasmtime
-        self.wait().await?;
         unimplemented!("There is currently no way to stop a running wasmtime instance")
     }
 
-    /// Returns the current status of the process
-    pub async fn status(&self) -> anyhow::Result<ContainerStatus> {
-        // NOTE: For those who modify this in the future, borrow must be as
-        // short lived as possible as it can block the send half. We do not use
-        // the recv method because it uses the value each time and then waits
-        // for a new value on the next call, whereas we want to return the last
-        // sent value until updated
-        Ok(self.status_channel.borrow().clone())
+    pub(crate) fn status(&self) -> Receiver<ContainerStatus> {
+        self.status_channel.clone()
     }
 
-    // For now this is private (for use in testing and in stop). If we find a
-    // need to expose it, we can do that later
     pub(crate) async fn wait(&mut self) -> anyhow::Result<()> {
         (&mut self.handle).await.unwrap()
     }
 }
+
+/// PodHandle is the top level handle into managing a pod. It manages updating
+/// statuses for the containers in the pod and can be used to stop the pod and
+/// access logs
+pub struct PodHandle<R: AsyncReadExt + AsyncSeekExt + Unpin> {
```

Comment:
I don't think you can just use `AsyncRead + AsyncSeek`

Reply:
I think I need Ext here as well. As in the previous PR, I don't think you can use the methods I do without those traits in scope